tree/tree/src/TChain.cxx

```diff
@@ -68,38 +68,34 @@ ClassImp(TChain);
 ////////////////////////////////////////////////////////////////////////////////
 /// Default constructor.
 
-TChain::TChain()
-: TTree()
-, fTreeOffsetLen(100)
-, fNtrees(0)
-, fTreeNumber(-1)
-, fTreeOffset(0)
-, fCanDeleteRefs(kFALSE)
-, fTree(0)
-, fFile(0)
-, fFiles(0)
-, fStatus(0)
-, fProofChain(0)
+TChain::TChain(bool locallyManaged)
+   : TTree(), fTreeOffsetLen(100), fNtrees(0), fTreeNumber(-1), fTreeOffset(0), fCanDeleteRefs(kFALSE), fTree(0),
+     fFile(0), fFiles(0), fStatus(0), fProofChain(0), fLocallyManaged(locallyManaged)
 {
    fTreeOffset = new Long64_t[fTreeOffsetLen];
    fFiles = new TObjArray(fTreeOffsetLen);
    fStatus = new TList();
    fTreeOffset[0]  = 0;
-   if (gDirectory) gDirectory->Remove(this);
-   gROOT->GetListOfSpecials()->Add(this);
+   if (!fLocallyManaged) {
+      if (gDirectory)
+         gDirectory->Remove(this);
```

Comment:
Should `gDirectory` still memory manage `*this` if `fLocallyManaged`?

Reply:
Probably not.  But actually I missed skipping the ```Append``` side of this, which occurs in the TTree constructor (and is already marked as a FIXME: https://github.com/root-project/root/blob/04375696c2e00be2c8881636e9ac9d43d7d55e54/tree/tree/src/TTree.cxx#L899-L903)

I would propose to skip both the ```Append``` and ```Remove``` if ```fLocallyManaged```

Reply:
Actually, I think the `gDirectory->Remove(this);` is now unnecessary (as `TTree::TTree()` no longer add it).

Reply:
And note:t that the code in TTree.cxx:903 should be reacheable from the TChain constructor.

Reply:
You mean should *not* be reachable from the TChain constructor right?  (because it calls the TTree constructor with ```dir = nullptr```)

Reply:
If I am not mistaken, `TChain`'s constructor use the default `TTree` constructor which does 'nothing'.

Reply:
The second ```TChain``` constructor (also relevant here) calls the non-default ```TTree``` constructor.  (But as said the ```gDirectory->Append``` call is anyways skipped because it's called with ```dir = nullptr```, so we can still skip the ```gDirectory->Remove``` call here.)